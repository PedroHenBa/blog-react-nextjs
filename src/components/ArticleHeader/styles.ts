import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xbig};
    margin-bottom: ${theme.spacing.xbig};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.spacing.medium};
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `}
`;
