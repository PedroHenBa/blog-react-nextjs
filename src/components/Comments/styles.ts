import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacing.big} auto;
    padding: 0 ${theme.spacing.big};
    color: ${theme.colors.darkText};
  `}
`;
