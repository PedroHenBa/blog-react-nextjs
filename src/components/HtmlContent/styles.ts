import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.medium};

    p {
      margin: ${theme.spacing.big} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      transition: all 200ms ease-in-out;
    }

    a:hover {
      filter: brightness(70%);
    }

    code {
      font-family: monospace;
      color: ${theme.colors.secondary};
      font-size: ${theme.fonts.sizes.small};
      background: ${theme.colors.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.primary};
      padding: ${theme.spacing.medium};
      color: ${theme.colors.white};
      font-family: monospace;
      margin: ${theme.spacing.medium} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.fonts.sizes.small};
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacing.medium} 0;
    }

    .image figcaption {
      padding: ${theme.spacing.small};
      text-align: center;
      font-size: ${theme.fonts.sizes.small};
      line-height: 1.3;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacing.medium} 0;
    }

    ul,
    ol {
      margin: ${theme.spacing.medium} ${theme.spacing.xbig};
    }

    hr {
      border: none;
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      border-collapse: collapse;
      border: 1px solid #eee;
      border-bottom: 2px solid #00cccc;
      box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.1), 0px 10px 20px rgba(0, 0, 0, 0.05), 0px 20px 20px rgba(0, 0, 0, 0.05),
        0px 30px 20px rgba(0, 0, 0, 0.05);
      tr {
        &:hover {
          background: #f4f4f4;

          td {
            color: #555;
          }
        }
      }
      th,
      td {
        color: #999;
        border: 1px solid #eee;
        padding: 12px 35px;
        border-collapse: collapse;
      }
      th {
        background: #00cccc;
        color: #fff;
        text-transform: uppercase;
        &.last {
          border-right: none;
        }
      }
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkGray};
      filter: brightness(80%);
      padding-left: ${theme.spacing.medium};
      font-style: italic;
      margin: ${theme.spacing.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
