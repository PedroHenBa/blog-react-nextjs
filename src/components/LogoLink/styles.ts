import styled from 'styled-components';

export const Container = styled.a`
  display: flex;
  color: inherit;
  text-decoration: none;
  align-items: center;

  > img {
    width: 15rem;
    height: 15rem;
    border-radius: 50%;
  }
`;
