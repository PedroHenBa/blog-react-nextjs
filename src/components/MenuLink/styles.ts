import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    color: ${theme.colors.white};
    margin-bottom: ${theme.spacing.small};
    font-size: 1.8rem;
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 300ms ease-in-out;

    &:hover {
      border-right: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.secondary};
    }
  `}
`;
