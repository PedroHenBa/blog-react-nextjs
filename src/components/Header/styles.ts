import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacing.xhuge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacing.small} - 1rem);
    }

    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeadingStyles} {
        margin: 0 0 ${theme.spacing.medium} 0;
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    flex-flow: row wrap;
    margin-left: ${theme.spacing.medium};
    max-width: 44rem;

    @media ${theme.media.lteSmall} {
      margin-left: 0;
    }
  `}
`;
