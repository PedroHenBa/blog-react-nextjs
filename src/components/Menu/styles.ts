import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export type MenuOpen = {
  isMenuOpen: boolean;
};

const styledMenuIcon = (theme: DefaultTheme, isMenuOpen: MenuOpen['isMenuOpen']) => {
  if (isMenuOpen) {
    return css`
      color: ${theme.colors.secondary};
      left: 26rem;
    `;
  } else {
    return css`
      color: ${theme.colors.white};
      left: 0rem;
    `;
  }
};

export const Container = styled.div<MenuOpen>`
  ${({ theme, isMenuOpen }) => css`
    z-index: 1;
    position: fixed;
    padding: ${theme.spacing.big};
    background: ${theme.colors.primary};
    height: 100vh;
    width: 100%;
    max-width: 32rem;
    display: flex;
    top: 0;
    flex-flow: column nowrap;
    left: ${isMenuOpen ? 0 : '-31rem'};
    overflow-y: ${isMenuOpen ? 'auto' : 'hidden'};
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      left: ${isMenuOpen ? 0 : '-32rem'};
    }
  `}
`;

export const MenuOpen = styled.a<MenuOpen>`
  ${({ theme, isMenuOpen }) => css`
    z-index: 2;
    position: fixed;
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    width: 3rem;
    height: 3rem;
    top: ${theme.spacing.medium};
    ${styledMenuIcon(theme, isMenuOpen)}
    transition: all 300ms ease-in-out;
  `}
`;

export const NavLinks = styled.nav`
  margin: auto;
  width: 100%;
`;

export const NavLogo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacing.xxbig};
      img {
        border: 0.5rem solid ${theme.colors.secondary};
      }
    }
  `}
`;
