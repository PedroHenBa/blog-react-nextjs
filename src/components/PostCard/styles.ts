import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacing.small} - 0.6rem);
      margin-bottom: ${theme.spacing.small};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    &:hover a {
      color: ${theme.colors.secondary};
      filter: brightness(70%);
    }

    &:hover img {
      filter: blur(1px) brightness(100%);
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
`;

export const Excerpt = styled.p``;
