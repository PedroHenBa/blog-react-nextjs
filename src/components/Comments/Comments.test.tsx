import { Comments, CommentsProps } from './index';
import { renderTheme } from '../../styles/render-theme';

const props: CommentsProps = {
  id: '1',
  title: 'Salve',
  slug: 'salve',
  allowComments: true,
};

describe('<Comments />', () => {
  it('should match snapshot with area for comments', () => {
    const { container } = renderTheme(<Comments {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render area for comments', () => {
    const { container } = renderTheme(<Comments {...props} allowComments={false} />);
    expect(container.firstChild).toBe(null);
  });
});
