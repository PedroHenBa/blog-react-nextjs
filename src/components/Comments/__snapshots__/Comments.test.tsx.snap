// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> should match snapshot with area for comments 1`] = `
.c0 {
  max-width: 80rem;
  width: 100%;
  margin: 3.2rem auto;
  padding: 0 3.2rem;
  color: #333333;
}

<div
  class="c0"
>
  <div
    id="disqus_thread"
  />
</div>
`;
